calabash-cucumber/lib/calabash-cucumber/logging.rb

```diff
@@ -74,6 +100,27 @@ def self.cyan(string)
     def self.green(string)
       colorize(string, 32)
     end
+
+    # @!visibility private
+    def self.timestamp
+      Time.now.strftime("%Y-%m-%d_%H-%M-%S")
+    end
+
+    # @!visibility private
+    def self.logs_directory
+      path = File.join(Calabash::Cucumber::DotDir.directory, "logs")
+      FileUtils.mkdir_p(path)
+      path
+    end
+
+    # @!visibility private
+    def self.calabash_log_file
+      path = File.join(self.logs_directory, "calabash.log")
+      if !File.exist?(path)
```

Comment:
You know, I was thinking of you Tobias when I wrote that.  Last summer I recall you mentioning that you preferred `!` over `unless`.  I can go either way.